docs/interpreter_design_doc.md

```diff
@@ -0,0 +1,100 @@
+# Interpreter Design
+
+
+## Class Hierarchy
+
+Stablehlo programs are computations over Tensors (n-dimensional arrays), hence
```

Comment:
I'd qualify this sentence with "at the moment", e.g. "At the moment, StableHLO programs are computations over tensors". To represent programs with dynamic shapes, we'll need to add support for shape computations, and they may not necessarily use tensors.